src/ledger/backend/tokencache.ts

```diff
@@ -20,10 +20,31 @@ interface Responder {
 	symbol(): Promise<string>;
 }
 
+export interface TokenCache {
+	readonly typeCache: TokenTypes;
+	readonly holderCache: TokenHolders;
+	setType(tokenAddr: string, ttype: TokenType): void;
+	tokenHolderFor(erdstall: Erdstall, ttype: TokenType): Promise<string>;
+	tokenTypeOf(erdstall: Erdstall, tokenAddr: string): Promise<TokenType>;
+	findRegisteredTokenWithSymbol(
+		erdstall: Erdstall,
+		symbol: string,
+		fromBlock?: number,
+	): Promise<string | undefined>;
+	queryRegisteredTokenTypes(
+		erdstall: Erdstall,
+		fromBlock?: number,
+	): Promise<TokenTypeRegisteredEvent[]>;
+	queryRegisteredTokens(
+		erdstall: Erdstall,
+		fromBlock?: number,
+	): Promise<TokenRegisteredEvent[]>;
+}
+
 type TokenTypes = Map<string, TokenType>;
 type TokenHolders = Map<TokenType, string>;
 
-export class TokenTypesCache {
+export class TokenTypesCache implements TokenCache {
```

Comment:
Hmm, I find the naming right now somewhat confusing. Maybe rename the interface to something like `TokenProvider` and the implementation to something like `TokenFetcher`?

Reply:
Naming is hard and I really prefer your suggestion.